pkg/proxy/iptables/proxier.go

```diff
@@ -1300,7 +1304,10 @@ func (proxier *Proxier) syncProxyRules() {
 				)
 				if !svcInfo.NodeLocalExternal() {
 					// Nodeports need SNAT, unless they're local.
+					// Temporarily replacing the chain name (args[1]) to which we append the rule.
+					args[1] = string(svcChain)
 					utilproxy.WriteLine(proxier.natRules, append(args, "-j", string(KubeMarkMasqChain))...)
+					args[1] = string(kubeNodePortsChain)
 					// Jump to the service chain.
 					utilproxy.WriteLine(proxier.natRules, append(args, "-j", string(svcChain))...)
```

Comment:
so actually, another possibility would be to add a `utilproxy.WriteRuleLine` helper so that then you can construct `args` without the `-A` arg, and then here you'd do

    utilproxy.WriteRuleLine(proxier.natRules, svcChain, append(args, "-j", string(KubeMarkMasqChain))...)
    utilproxy.WriteRuleLine(proxier.natRules, kubeNodePortsChain, append(args, "-j", string(svnChain))...)

and it would prepend `{"-A", svcChain}` or `{"-A", kubeNodePortsChain}` and then pass it to `utilproxy.WriteLine`

Reply:
done.